src/Data/Text/Internal/Fusion.hs

```diff
@@ -98,6 +99,35 @@ stream (Text arr off len) = Stream next off (betweenSize (len `shiftR` 2) len)
               _ -> U8.chr4 n0 n1 n2 n3
 {-# INLINE [0] stream #-}
 
+-- | /O(n)/ @'streamLn' t = 'stream' (t <> \'\\n\')@
+--
+-- @since 2.1.2
+streamLn ::
+#if defined(ASSERTS)
+  HasCallStack =>
+#endif
+  Text -> Stream Char
+streamLn (Text arr off len) = Stream next off (betweenSize (len `shiftR` 2) (len + 1))
+    where
+      !end = off+len
+      next !i
+          | i > end = Done
+          | i == end  = Yield '\n' (i + 1)
```

Comment:
This is almost an exact copy of `stream`. A boolean argument `appendNewline` should be added so that there is not duplication.